pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -113,9 +120,72 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .filter(t -> (t.getInserted().matches("[a-zA-Z/]")))
+                           .subscribe(t -> {
+                               try {
+                                   if (xpathExpressionArea.getText().contains("/")) {
+                                       if (!t.getInserted().equals("/")) {
+                                           autoComplete(xpathExpressionArea.getText().substring(xpathExpressionArea.getText().lastIndexOf("/")));
+                                       }
+                                   }
+                               } catch (IOException e) {
+                                   e.printStackTrace();
+                               } catch (ClassNotFoundException e) {
+                                   e.printStackTrace();
+                               }
+                           });
+    }
+
+    private void autoComplete(String s) throws IOException, ClassNotFoundException {
+
+        autoCompletePopup = new ContextMenu();
```

Comment:
You still create a new context menu each time!

Reply:
Will make the update, sorry for the confusion before